pkgs/tools/X11/libstrangle/default.nix

```diff
@@ -23,6 +23,10 @@ stdenv.mkDerivation rec {
     substituteAllInPlace src/strangle.sh
     substituteAllInPlace src/stranglevk.sh
   '';
+  postInstall = ''
+    substituteInPlace $out/share/vulkan/implicit_layer.d/libstrangle_vk.json --replace "libstrangle_vk.so" "$out/lib/libstrangle/\$LIB/libstrangle_vk.so"
+    ln -s lib64 $out/lib/libstrangle/lib # $LIB above expands to lib but should expand to lib32 on 32bit if that ever becomes relevant
```

Comment:
I'm not sure if `$LIB` should ever expand to anything other than `lib` with the way Nix is designed. If we wanted to support this, we'd either have to have to hard-code all libraries in architecture-dependent lib directories (which would be redundant for most cases) or have separate builds for 32bit packages that run on 32bit devices vs. 64bit devices. I also mentioned this in https://github.com/NixOS/nixpkgs/issues/101597#issuecomment-834615474.

As a workaround to get the 32bit lib working, you can do something similar to what I've done for [mangohud](https://github.com/NixOS/nixpkgs/blob/c2f68f081c92d51df57154522a283739ab724a49/pkgs/tools/graphics/mangohud/default.nix#L130-L135) and [vkBasalt](https://github.com/NixOS/nixpkgs/blob/c2f68f081c92d51df57154522a283739ab724a49/pkgs/tools/graphics/vkBasalt/default.nix#L29-L33). Since https://github.com/KhronosGroup/Vulkan-Loader/pull/525, vulkan-loader will automatically select the layer that matches the appropriate architecture.